pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -158,13 +163,20 @@ func (c *FlowContext) ensureManagedVpc(ctx context.Context) error {
 		Tags:                         c.commonTags,
 		EnableDnsSupport:             true,
 		EnableDnsHostnames:           true,
-		AssignGeneratedIPv6CidrBlock: c.config.DualStack != nil && c.config.DualStack.Enabled,
+		AssignGeneratedIPv6CidrBlock: (c.config.DualStack != nil && c.config.DualStack.Enabled) || sets.New[v1beta1.IPFamily](c.ipFamilies...).Has(v1beta1.IPFamilyIPv6),
 		DhcpOptionsId:                c.state.Get(IdentifierDHCPOptions),
 	}
-	if c.config.Networks.VPC.CIDR == nil {
+
+	if isIPv4(c.ipFamilies) && c.config.Networks.VPC.CIDR == nil {
 		return fmt.Errorf("missing VPC CIDR")
 	}
-	desired.CidrBlock = *c.config.Networks.VPC.CIDR
+
+	if !isIPv4(c.ipFamilies) && c.config.Networks.VPC.CIDR == nil {
+		desired.CidrBlock = "10.0.0.0/16"
```

Comment:
I think I forgot the outcome of that discussion around this hardcoded IP and it would be nice if here or better in a comment write something about it.

Reply:
Probably, it is better not to automatically assign an IPv4 address here and leave that to the user?

Reply:
Though, I need to adapt the validation.